OEP-6/OEP-6.mediawiki

```diff
@@ -0,0 +1,394 @@
+
+===Network===
+A network API consists of:
+
+<pre>
+type NetworkType = 'MAIN' | 'TEST' | 'PRIVATE';
+type Asset = 'ONT' | 'ONG';
+
+interface Network {
+  type: NetworkType;
+  address: string;
+}
+
+function getGenerateBlockTime(): Promise<number | null>
+function getNodeCount(): Promise<number>
+function getBlockHeight(): Promise<number>
+function getMerkleProof({ txHash: string }): Promise<MerkleProof>
+function getStorage({ contract: string, key: string }): Promise<string>
+function getAllowance({ asset: Asset, fromAddress: string, toAddress: string }): Promise<number>
+function getBlock({ block: number | string }): Promise<Block>
+function getTransaction({ txHash: string }): Promise<Transaction>
+function getNetwork(): Network
+function getBalance({ address: string }): Promise<Balance>
+</pre>
+
+For further explanation about the wrapped method consult https://ontio.github.io/documentation/restful_api_en.html . The types '''Transaction''', '''Block''', '''MerkleProof''' and '''Balance''' corresponds to the exact object returned from Ontology blockchain.
+
+TODO: copy the definition of all network api calls with inputs and outputs from ontology documentation.
+
+===Provider===
+A primary focus of Provider API is to check if user has installed any dAPI provider and to get information about it.
+
+====getProvider====
+<pre>
+
+interface Provider {
+  name: string;
+  version: string;
+  compatibility: string[];
+}
+
+function getProvider(): Promise<Provider>
+</pre>
+
+Returns the information about installed dAPI provider if there is one installed. <code>compatibility</code> refers to all OEPs this dAPI provider supports. E.g.:
+
+<pre>
+compatability: [
+    'OEP-6',
+    'OEP-10',
+    'OEP-29'
+]
+</pre>
+
+dAPI provider does not need to support all future OEPs concerned with dAPI.
+
+* Rejects with '''NO_PROVIDER''' in case there is no dAPI provider installed.
+
+===Asset===
+A primary focus of Asset API is to initiate a transfer. The request needs to be confirmed by the user.
+
+====getAccount====	
+<pre>	
+function getAccount(): Promise<string>	
+</pre>	
+Returns currently selected account of logged in user. 	
+ * Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+
+====makeTransfer====
+<pre>
+function makeTransfer({ recipient: string, asset: Asset, amount: number }): Promise<string>
+</pre>
+
+Initiates a transfer of <code>amount asset</code> to <code>recipient</code> account. 
+
+The amount should always be an integer value. Therefore is is represented in the smallest, further non divisible units for the asset. E.g.: If the <code>asset</code> is ONG then the value is multiplied by 1000000000.
+
+Returns transaction Id.
+
+* Rejects with '''NO_ACCOUNT''' in case the user is not signed in or has no accounts
+* Rejects with '''MALFORMED_ACCOUNT''' in case the <code>recipient</code> is not a valid account
+* Rejects with '''CANCELED''' in case the user cancels the request
```

Comment:
Will we also need an error in the case that the transaction is rejected?

Reply:
It depends. I can add it here, but I don't want to go any further as to distinguishing "not enough amount", ... because it will reveal too much information. But a simple FAILED/REJECTED will do.